scripts/host-backup-restore/host-backup

```diff
@@ -23,11 +29,19 @@ tar --exclude 'tmp/*' --exclude 'tmp/.*' --exclude 'var/crash/*' --exclude 'var/
     --exclude 'var/log/audit*' \
     --exclude 'var/xsconfig/*' --exclude 'var/xsconfig/.*' \
     --exclude 'var/xapi/[0-9a-f]*-[0-9a-f]*-[0-9a-f]*-[0-9a-f]*-[0-9a-f]*' \
+    --exclude-from ${EXCLUDE_FILE} --exclude ${EXCLUDE_FILE} \
     --sparse --preserve-permissions --to-stdout --gzip --one-file-system -C / -c . 
 
-# Remove the backup of the pool-database.  We only need it to be present in the tar file.
+TAR_EXIT_STATUS=$?
+
+# Remove the backup of the pool-database and the exclude list.
+# We only need it to be present in the tar file.
 # During restore the tar file is extracted over the backup partition which is then made
 # bootable.  Following a reboot the user must manually call "xe pool-restore-database" on 
 # the pool-database backup file.
 
-rm -f ${POOL_DB_BACKUP}
+rm -f ${POOL_DB_BACKUP} ${EXCLUDE_FILE}
```

Comment:
If you use the `--warning=...` suggestion above, you don't need this.   If you still need this, you can ensure that the files are removed by putting these lines in an exit trap and still using `setopt -e` to make bash bail out on failure.

https://redsymbol.net/articles/bash-exit-traps